scripts/helper_numpy.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+# coding: utf8
+'''Compute a Numpy friendly data-structures from VGGish files
+Example
+-------
+$ cd {repo_root}
+$ ./scripts/helper_numpy.py --csv_file /path/to/sparse-labels.csv
+--vggish_path /path/to/vggish/ --output_file /path/to/output.npz
+
+Any valid output file '*.npz' has to be speficied by the user
+'''
+
+import argparse
+import numpy as np
+import os
+import pandas as pd
+import sys
+import json
+
+
+def main(csvfile, vggishpath, outfile):
+
+    success = []
+    df = pd.read_csv(csvfile)
+    instruments = np.unique(df['instrument'])
+    sample_key = np.unique(df['sample_key'])
+
+    print('Extracting the vggish features...')
+    X = np.empty([20000, 10, 128], dtype=int)
+    count = 0
+    for sk in sample_key:
+        sk_prefix = sk[:3]
+        full_name = vggishpath + sk_prefix + '/' + sk + '.json'
+        with open(full_name, 'r') as f:
+            X_tmp = json.load(f)
+        X[count] = X_tmp['features']
+        count += 1
+
+    print('Extracting the labels information...')
+    Y_true = 0.5 * np.ones([20000, 20], dtype=float)
+    Y_mask = np.zeros([20000, 20], dtype=bool)
+
+    inst_lab = np.copy(df)
+    for inst in inst_lab:
+        x_pos = int(np.arange(20000)[sample_key == inst[0]])
+        y_pos = int(np.arange(20)[instruments == inst[1]])
+        Y_true[x_pos, y_pos] = 0.5 * (inst[2] + 1)  # the conversion is probably temporary
+        # until we get the sparse labels in the right format
+        Y_mask[x_pos, y_pos] = 1
+
+    print('Saving the NPZ file...')
+    try:
+        np.savez(outfile, X=X, Y_true=Y_true,
+                 Y_mask=Y_mask, sample_key=sample_key)
+
+        success.append(os.path.exists(outfile))
+    except ValueError:
+        pass
```

Comment:
what's the thinking here on catching the ValueError? if something is wonky, I'm fine to let it fail hard?

Reply:
You mean remove the try statement completely?

Reply:
yea, generally catching Errors / Exceptions and silencing them completely isn't great practice. Other than letting the error happen, you could catch it and warn that something is bad, like 

```python
try: 
    ...something...
except ValueError as uhoh:
    warnings.warn('caught a thing but continuing: {}'.format(uhoh))
    success = False
```

or catch it, do something, and re-raise:

```python
try: 
    ...something...
except ValueError as uhoh:
    print("did you mean foo?")
    raise uhoh
```